src/linker/Linker.Steps/RootNonTrimmableAssemblies.cs

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+
+namespace Mono.Linker.Steps
+{
+	public class RootNonTrimmableAssemblies : BaseStep
```

Comment:
I don't understand why this step is needed?

Reply:
It's needed for example if someone passes `-reference UnreferencedAssembly --action copy UnreferencedAssembly` - then we need to make sure the unreferenced assembly is kept. This could happen if someone has an unused nuget package dependency (or one that's used via non-understood reflection).

Reply:
Why would we want to do that? It would keep unused assembly which someone redundantly passed as an argument. We have `-a` option which does the same thing and should be used if it's intentional.

Reply:
Maybe a better example is `-reference UnreferencedAssembly --action copy` (setting the *default* action to copy instead). In this case we would need to root the assembly if it isn't attributed as `IsTrimmable`.
- If someone publishes an app with a (statically unreferenced) nuget dependency, *without* trimming, they would expect to see it in the output. Any non-`IsTrimmable` assemblies should similarly be kept with `PublishTrimmed`.
- We don't want to pass `-a` for such cases because it would prevent trimming for dependencies that *are* `IsTrimmable`.

Then I would do the same with the per-assembly `copy` action for consistency (this is the first suggestion I made in https://github.com/mono/linker/issues/1800#issuecomment-773497902).